python/pythonlab/neighborhood/neighborhood/support/neighborhood_signal_message.py

```diff
@@ -1,26 +1,35 @@
 import json
 from .signal_message_type import SignalMessageType
+from .neighborhood_context_type import NeighborhoodContextType
+from .neighborhood_signal_key import NeighborhoodSignalKey
 
 class NeighborhoodSignalMessage:
     """
     All Neighborhood signal messages will use the following format:
         {
             "type": <SignalMessageType>,
-            "value": <"message data">,
+            "key": <NeighborhoodSignalKey>,
             "detail": <dict>
         }
     """
 
-    def __init__(self, type: SignalMessageType, value: str, detail: dict = None):
+    def __init__(self, type: SignalMessageType, key: NeighborhoodSignalKey, detail: dict):
         self.type = type
-        self.value = value
+        self.key = key
         self.detail = detail
 
-    def get_formatted_message(self) -> str:
+    def _get_formatted_message(self) -> str:
         """
         @return: A string representing the neighborhood signal message
         """
-        formatted_message = f'[{self.type.value}] {self.value}'
+        formatted_message = f'[{self.type.value}] {self.key.value}'
         if self.detail:
             formatted_message += f' {json.dumps(self.detail)}'
         return formatted_message
+
+    def send(self, world_context_type, neighborhood_tracker, is_boolean_message):
+        if world_context_type == NeighborhoodContextType.RUN and not is_boolean_message:
+            print (self._get_formatted_message())
+        elif world_context_type == NeighborhoodContextType.VALIDATE:  
+            if neighborhood_tracker:
```

Comment:
could we just get the neighborhood tracker here rather than passing it in?

Reply:
Mentioned this above, but I am passing the neighborhood tracker instance here bc when I imported NeighborhoodTracker in neighborhood_signal_message, I got a circular reference error.
I will add a comment here - meant to do that!

Reply:
makes sense!